pubtools/_pulp/services/pulp.py

```diff
@@ -87,9 +98,13 @@ def pulp_fake_controller(self):
 
     def new_pulp_client(self, **kwargs):
         """Creates and returns a new Pulp client with appropriate config."""
+        cert = None
         auth = None
         args = self._service_args
 
+        # Use hook to get the certificate path(s). Defined here, used later.
+        hook_rets = pm.hook.get_cert_key_paths(server_url=args.pulp_url)
```

Comment:
I looked at the implementation of this hook (in Pub) and it is unfortunately implemented to *raise an exception*, rather than return None, if no cert/key has been deployed.

That's going to cause a problem then, because we haven't deployed certs/keys yet (since we haven't implemented support for them in Pulp yet).  So as soon as this code is deployed onto Pub it'll start crashing.

We need to handle that somehow, and it should be handled before this change is submitted (otherwise pubtools-pulp becomes effectively unreleasable).

Given that the hook documentation at https://release-engineering.github.io/pubtools/hooks.html#get_cert_key_paths says that "Callers of this hook should be prepared to receive no result", I'd say it's the implementation in Pub which is wrong, as it will never return no result (you will always either get a client cert & key, or get an exception).

Would you be able to resolve that?  Either changing the hook in Pub so that it logs a warning and returns None instead of raising; or ensuring that we actually deploy some certs at the expected paths.